js/layout-javascript/news-feed-code.js

```diff
@@ -2480,6 +2480,16 @@ DynamicList.prototype.addDetailViewData = function(entry) {
     var content = '';
 
     if (obj.type === 'file') {
+      if (files && Array.isArray(files)) {
```

Comment:
@dwfliplet do we have a shared JS file where we can define a function for this code? We could then utilize this function across these three files to adhere to the DRY principle.

Reply:
@armine-fliplet Generally speaking, I agree that would be sufficient, but a big part of those files are duplicated. For example, the rest of the code is almost identical or identical in all three files. So there's no point in extracting a few lines when they're part of a bigger repeated sequence. To correctly manage duplications in those templates, I'd need to refactor those three files (>2k, >3k, and >3k lines), which I believe is out of the scope of this bugfix 😄.

But you're right, @armine-fliplet, that code needs refactoring in terms of duplications in order to decrease the amount of code to maintain. I think it's worth logging that in backlog, to plan maintenance in proper time. Wdyt @simonchapman1986?

Reply:
So firstly, great to see your desire of keeping code dry @armine-fliplet - its certainly something we want and need.

However, I do think what @dwfliplet makes sense in this context - we shouldn't bring into scope a large refactor as this is a bugfix. If this was feature dev, i'd be more inclined to say - lets do the refactor. But we want to get this out the door quickly and efficiently.

I would say its worth adding a low priority item for the backlog noting this however, so we can indeed give it some time when we feel appropriate.